resource/environment/integration.js

```diff
@@ -201,6 +206,14 @@ function configure( envelope ) {
 
 function upgrade( slack, envelope ) {
 	var image = envelope.data.image;
+	if ( !util.checkInfo( util.getImageInfo( image ) ) ) { //check tag if tag is formated correctly
+		return {
+			status: 500,
```

Comment:
I think this should be a 400 status instead of a 500. In this path the client is trying to execute a PUT for the resource (a docker image) and we're saying that the syntax of the image is bad (i.e. it is not formatted by build goggles)